hashintel/hash
# Expose ontology type retrieval in temporal worker

## 🌟 What is the purpose of this PR?

The Graph API is currently only available from the TS backend. In order to also make it available as a temporal service (e.g. in the Python workers) without the need to create new bindings this exposes three functions to read ontology types similar to the interface in `hash-api`.

It's possible to also generate bindings for Python, but this way it's possible to re-use functions which are already implemented in Typescript such as `getRoots` or the "insert-or-load" behavior.

## 🔗 Related links





- Part of [this Asana task](https://app.asana.com/0/1204727813960182/1204904869169432/f) _(internal)_

## 🚫 Blocked by

- https://github.com/hashintel/hash/pull/2693
- https://github.com/hashintel/hash/pull/2696

## 🔍 What does this change?

- Remove old example activity/workflow
- Add activities to read ontology types
- Expose workflow to use these activities
## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?

This PR:


### 📜 Does this require a change to the docs?

The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:


## 🐾 Next steps

Adjust the Python worker to use these workflows

## 🛡 What tests cover this?

This is currently not tested automatically

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Start external services (`temporal-setup`, `hash-temporal-worker-ts`, `graph`)
1.  Confirm the following workflow returns the (recursive) person entity type:

    ```
    temporal workflow execute -t ai --type getEntityType -i '{ "entityTypeId": "https://blockprotocol.org/@examples/types/entity-type/person/v/1"}'
    ```

## 📹 Demo

```
Running execution:
  WorkflowId  218425e3-7810-4c5a-833a-6b30f33ccaf7
  RunId       2a7779cb-9305-4ca0-a178-5bba2b9523bc
  Type        getEntityType
  Namespace   default
  TaskQueue   ai
  Args        [{"entityTypeId":"https://blockprotocol.org/@examples/types/entity-type/person/v/1"}]

Progress:

Time elapsed: 1s
  ID          Time                     Type
   1  2023-06-29T12:33:52Z  WorkflowExecutionStarted
   2  2023-06-29T12:33:52Z  WorkflowTaskScheduled
   3  2023-06-29T12:33:52Z  WorkflowTaskStarted
   4  2023-06-29T12:33:52Z  WorkflowTaskCompleted
   5  2023-06-29T12:33:52Z  ActivityTaskScheduled
   6  2023-06-29T12:33:52Z  ActivityTaskStarted
   7  2023-06-29T12:33:54Z  ActivityTaskCompleted
   8  2023-06-29T12:33:54Z  WorkflowTaskScheduled
   9  2023-06-29T12:33:54Z  WorkflowTaskStarted
  10  2023-06-29T12:33:54Z  WorkflowTaskCompleted
  11  2023-06-29T12:33:54Z  WorkflowExecutionCompleted

Result:
  Run Time: 2 seconds
  Status: COMPLETED
  Output: [{"schema":{"$schema":"https://blockprotocol.org/types/modules/graph/0.3/schema/entity-type","kind":"entityType","$id":"https://blockprotocol.org/@examples/types/entity-type/person/v/1","title":"Person","description":"An extremely simplified representation of a person or human being.","type":"object","properties":{"https://blockprotocol.org/@examples/types/property-type/e-mail/":{"$ref":"https://blockprotocol.org/@examples/types/property-type/e-mail/v/1"},"https://blockprotocol.org/@blockprotocol/types/property-type/name/":{"$ref":"https://blockprotocol.org/@blockprotocol/types/property-type/name/v/1"}},"required":["https://blockprotocol.org/@blockprotocol/types/property-type/name/"],"links":{"https://blockprotocol.org/@examples/types/entity-type/employed-by/v/1":{"type":"array","items":{"oneOf":[{"$ref":"https://blockprotocol.org/@examples/types/entity-type/company/v/1"}]},"minItems":0,"ordered":false}}},"metadata":{"recordId":{"baseUrl":"https://blockprotocol.org/@examples/types/entity-type/person/","version":1},"provenance":{"recordCreatedById":"6a94cb79-999c-4c0f-9b56-d0a0940b42d8"},"fetchedAt":"2023-06-29T12:33:53.377491000Z"}}]
```
